helm search error with prelease versions that contains upper case characters

running ```helm search local/foo --versions``` fails when any of the versions contains upper case characters in the prerelease identifier.

To reproduce, use the following index.yaml:
```yaml
apiVersion: v1
entries:
  foo:
  - created: 2017-10-05T14:09:16.836314814-04:00
    description: Sample
    digest: cc33d14057547854591b210b920dd67647690b6ee731a89f36d135fcc99f490f
    name: foo
    urls:
    - http://localhost:8879/foo-16.0.0-Alpha.tgz
    version: 16.0.0-Alpha    
  - created: 2017-10-05T14:09:16.836314814-04:00
    description: Sample
    digest: cc33d14057547854591b210b920dd67647690b6ee731a89f36d135fcc99f490f
    name: foo
    urls:
    - http://localhost:8879/foo-16.0.0-Beta.tgz
    version: 16.0.0-Beta       
generated: 2017-11-02T10:25:46.654821464-04:00
```

Running ```helm search local/foo --versions``` gives the following:
```
$ helm search local/foo --versions
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1340ebf]

goroutine 1 [running]:
k8s.io/helm/cmd/helm/search.scoreSorter.Less(0xc0427709c0, 0x2, 0x2, 0x1, 0x0, 0x20)
        /go/src/k8s.io/helm/cmd/helm/search/search.go:217 +0xff
k8s.io/helm/cmd/helm/search.(*scoreSorter).Less(0xc04308dea0, 0x1, 0x0, 0x18)
        <autogenerated>:3 +0x7a
sort.insertionSort(0x205aa60, 0xc04308dea0, 0x0, 0x2)
        /usr/local/go/src/sort/sort.go:29 +0x78
sort.quickSort(0x205aa60, 0xc04308dea0, 0x0, 0x2, 0x4)
        /usr/local/go/src/sort/sort.go:211 +0x1f2
sort.Sort(0x205aa60, 0xc04308dea0)
        /usr/local/go/src/sort/sort.go:220 +0x80
k8s.io/helm/cmd/helm/search.SortScore(0xc0427709c0, 0x2, 0x2)
        /go/src/k8s.io/helm/cmd/helm/search/search.go:193 +0x74
main.(*searchCmd).run(0xc042494280, 0xc042335400, 0x1, 0x2, 0x0, 0x0)
        /go/src/k8s.io/helm/cmd/helm/search.go:90 +0xaf
main.newSearchCmd.func1(0xc0422ea480, 0xc042335400, 0x1, 0x2, 0x0, 0x0)
        /go/src/k8s.io/helm/cmd/helm/search.go:61 +0x7c
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).execute(0xc0422ea480, 0xc042335080, 0x2, 0x2, 0xc0422ea480, 0xc042335080)
        /go/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:599 +0x3f0
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc04244a240, 0xc04203bb00, 0xc042202000, 0xc042202240)
        /go/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:689 +0x340
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).Execute(0xc04244a240, 0x3, 0x3)
        /go/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:648 +0x32
main.main()
        /go/src/k8s.io/helm/cmd/helm/helm.go:155 +0x7e
```
According to [SemVer](http://semver.org/) rules the prerelease identifier is allowed to contain upper case characters.

I'm getting a similar error, but only when the `-l` option is passed to `helm search`.
```
% helm search some/repo -l
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13020be]

goroutine 1 [running]:
k8s.io/helm/cmd/helm/search.scoreSorter.Less(0xc422032000, 0x11a0, 0x1400, 0x0, 0x9a1, 0xc42094f101)
        /go/src/k8s.io/helm/cmd/helm/search/search.go:221 +0x12e
k8s.io/helm/cmd/helm/search.(*scoreSorter).Less(0xc422ad76e0, 0x0, 0x9a1, 0x801)
        <autogenerated>:3 +0x73
sort.doPivot(0x2012c00, 0xc422ad76e0, 0x0, 0x11a0, 0xc42006fba8, 0x4100ac)
        /usr/local/go/src/sort/sort.go:125 +0x1c1
sort.quickSort(0x2012c00, 0xc422ad76e0, 0x0, 0x11a0, 0x1a)
        /usr/local/go/src/sort/sort.go:192 +0x80
sort.Sort(0x2012c00, 0xc422ad76e0)
        /usr/local/go/src/sort/sort.go:220 +0x79
k8s.io/helm/cmd/helm/search.SortScore(0xc422032000, 0x11a0, 0x1400)
        /go/src/k8s.io/helm/cmd/helm/search/search.go:193 +0x6d
main.(*searchCmd).run(0xc420330640, 0xc420298d40, 0x1, 0x2, 0x0, 0x0)
        /go/src/k8s.io/helm/cmd/helm/search.go:90 +0xa8
main.newSearchCmd.func1(0xc4201ce6c0, 0xc420298d40, 0x1, 0x2, 0x0, 0x0)
        /go/src/k8s.io/helm/cmd/helm/search.go:61 +0x75
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).execute(0xc4201ce6c0, 0xc420298ba0, 0x2, 0x2, 0xc4201ce6c0, 0xc420298ba0)
        /go/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:599 +0x3e9
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4203cc480, 0xc42029f680, 0xc42029fd40, 0xc4202bc000)
        /go/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:689 +0x339
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4203cc480, 0x3, 0x3)
        /go/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
        /go/src/k8s.io/helm/cmd/helm/helm.go:155 +0x77
%
```
```
Client: &version.Version{SemVer:"v2.7.0", GitCommit:"08c1144f5eb3e3b636d9775617287cc26e53dba4", GitTreeState:"clean"}
```

--- a/cmd/helm/search/search.go
+++ b/cmd/helm/search/search.go
@@ -146,11 +146,11 @@ func (i *Index) SearchLiteral(term string, threshold int) []*Result {
 	term = strings.ToLower(term)
 	buf := []*Result{}
 	for k, v := range i.lines {
-		k = strings.ToLower(k)
-		v = strings.ToLower(v)
-		res := strings.Index(v, term)
-		if score := i.calcScore(res, v); res != -1 && score < threshold {
-			parts := strings.Split(k, verSep) // Remove version, if it is there.
+		lk := strings.ToLower(k)
+		lv := strings.ToLower(v)
+		res := strings.Index(lv, term)
+		if score := i.calcScore(res, lv); res != -1 && score < threshold {
+			parts := strings.Split(lk, verSep) // Remove version, if it is there.
 			buf = append(buf, &Result{Name: parts[0], Score: score, Chart: i.charts[k]})
 		}
 	}

--- a/cmd/helm/search/search_test.go
+++ b/cmd/helm/search/search_test.go
@@ -91,10 +91,10 @@ var indexfileEntries = map[string]repo.ChartVersions{
 			},
 		},
 		{
-			URLs: []string{"http://example.com/charts/santa-maria-1.2.2.tgz"},
+			URLs: []string{"http://example.com/charts/santa-maria-1.2.2-rc-1.tgz"},
 			Metadata: &chart.Metadata{
 				Name:        "santa-maria",
-				Version:     "1.2.2",
+				Version:     "1.2.2-RC-1",
 				Description: "Three boat",
 			},
 		},

